R/graded_condition.R

```diff
@@ -29,15 +29,29 @@ conditionMessage.gradethis_graded <- function(c) {
   }
 }
 
-# Turn errors into `fail()`ures
+# Capture and handle errors, by default turning them into internal problem grades.
+# Use `fail_if_error()` to convert errors to failing grades.
 capture_errors <- function(expr, on_error = NULL) {
   if (is.null(on_error)) {
-    on_error <- function(e, that_env) {
-      # must wrap in ignore statement to retrieve fail object
-      ret <- capture_graded({
-        fail(conditionMessage(e))
-      })
-      rlang::return_from(that_env, ret)
+    on_error <- function(err, that_env) {
+      # get relevant calls from the stack to improve error reporting
+      calls <- sys_calls_most_helpful()
+      
+      # Rewrite the error call with the more helpful call
+      err$call <- calls$last
+      
+      if (!is.null(calls$first)) {
+        calls$first <- deparse(calls$first, getOption("width", 80))
+        err$call_gradethis <- paste(calls$first, collapse= "\n")
```

Comment:
Can we rename `$call_gradethis` to avoid a partial match to `$call` if `$call` is missing?

`$gradethis_call`?

Reply:
```suggestion
        err$gradethis_call <- paste(calls$first, collapse= "\n")
```